course_content/notebooks/dask_intro.ipynb

```diff
@@ -0,0 +1,216 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# An Introduction to Dask\n",
+    "\n",
+    "Dask is a flexible parallel computing library for Python. It allows dynamic scheduling of tasks for optimising computations, and includes collections such as arrays and lists for parallel data processing."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Terminology\n",
+    "\n",
+    "### General"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "#### Thread\n",
+    "A thread is the simplest unit of computation.\n",
+    "\n",
+    "#### Process\n",
+    "A process is an isolated computation which consists of one or more threads."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Dask\n",
+    "\n",
+    "#### Dask architecture\n",
+    "Below is the basic structure of how the components of dask interact:\n",
+    "\n",
+    "![](../images/dask_architecture.png)\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "#### Client\n",
+    "A client is a Python class that provides the primary point of access to a scheduler and its associated workers. "
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "#### Scheduler\n",
+    "One of Dask’s key benefits is its ability to efficiently schedule tasks to optimize computations. Everything Dask does is built on top of “schedulers”. These schedulers take the order of work established by a task graph and find the optimal way to break down and carry out the tasks.\n",
+    "\n",
+    "Dask has four types of schedulers:\n",
+    "- Synchronous: Single thread (good for debugging).\n",
+    "        dask.get\n",
+    "- Threaded: Utilises a thread pool.\n",
+    "        dask.threaded.get\n",
+    "- Multiprocessing: Utilises a process pool.\n",
+    "        dask.multiprocessing.get\n",
+    "- Distributed: Utilises a cluster of distributed machines.\n",
+    "        distributed.Client.get\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "#### Worker\n",
+    "A worker receives work from the scheduler to process and returns output to the scheduler."
```

Comment:
Receives **tasks** from the scheduler, perhaps? Prevents overloading the term "work".

Reply:
Yes, **tasks** is better.